Add orders fulfilled infobox

BlueSoapTurtle opened the pull request:
We're able to track the amount of orders fulfilled by checking the pvarbit 4480,
therefore we can add an infobox to display it.
the infobox will be disabled by default.

BlueSoapTurtle committed:
Add orders fulfilled infobox

We're able to track the amount of orders fulfilled, therefore we can
add an infobox to display it.
the infobox will be disabled by default.

hex-agon commented on src/main/java/work/fking/masteringmixology/MasteringMixologyPlugin.java:
```diff
@@ -394,6 +417,22 @@ private PotionModifier getPotionModifier(int orderIdx) {
+        if (config.showOrdersFulfilledInfobox()) {
+            int ordersFulfilled = client.getVarpValue(VARPBIT_ORDERS_FULFILLED);
+            if (ordersFulfilledInfoBox == null) {
+                BufferedImage fulfilledImage = ImageUtil.loadImageResource(getClass(), "orders_fulfilled.png");
```
Since you're using an image which is just a render of an item you can just get it straight from `ItemManager.getImage(itemId)` thus avoiding the need of bundling the image.

BlueSoapTurtle replied:
That's amazing! I'll do that later tonight and rebase it unless someone else has time before then.

Know if it's possible to get the icons from the ui? Like the blue mox icon below the orders panel, I know it has a sprite Id.

hex-agon replied:
`SpriteManager`

BlueSoapTurtle replied:
Thanks! Fixed the comment

hex-agon commented:
Also please rebase this PR

hex-agon commented on src/main/java/work/fking/masteringmixology/MasteringMixologyPlugin.java:
```diff
@@ -44,6 +48,8 @@ public class MasteringMixologyPlugin extends Plugin {
     private static final int VARBIT_POTION_ORDER_3 = 11319;
     private static final int VARBIT_POTION_MODIFIER_3 = 11320;
 
+    private static final int VARPBIT_ORDERS_FULFILLED = 4480;
```
Prefix this with `VARP_` also if you can, move it below the `_RESIN` constants and also rename them to `VARP_`, i mistakenly named them `VARBITS_`.

hex-agon replied:
```suggestion
    private static final int VARP_ORDERS_FULFILLED = 4480;
```

BlueSoapTurtle commented:
Fixed the comments and rebased the changes. Not sure if there's a better way than rebasing and force pushing to the branch on the forked repo but it's up to date at least :)

hex-agon commented:
If you can, please fix this checkstyle violation:
https://github.com/BlueSoapTurtle/mastering-mixology/actions/runs/11095008585/job/30823122369

GH actions are not triggering in the PR because it was opened before i added them i think

BlueSoapTurtle committed:
Fix code styles

BlueSoapTurtle commented:
Ready for review / merge
Will do a bit better with the commits in future PRs, new to github PRs

hex-agon commented on src/main/java/work/fking/masteringmixology/MasteringMixologyConfig.java:
```diff
@@ -11,6 +11,7 @@
 
 @ConfigGroup(CONFIG_GROUP)
 public interface MasteringMixologyConfig extends Config {
+    String GROUP = "masteringmixology";
```
Duplicate group declaration

BlueSoapTurtle replied:
Removed the duplicate string and made it remove the infobox when the player leaves the area

hex-agon reviewed (commented):
I think you should also remove the infoboxes whenever the player either:
- Leaves the mixology lab (use the widgetclosed event)
- Leaves the lab dungeon itself

This way we don't have 'rogue' infoboxes for players that leave the area.

raiyni commented:
Needs rebased and feedback addressed

BlueSoapTurtle committed:
Merge branch 'master' into add-order-count-infobox

BlueSoapTurtle commented:
back home now so will update the PRs, just rebased it but will fix the comments soon

BlueSoapTurtle committed:
Fix checkstyle

BlueSoapTurtle committed:
Hide infobox outside the labs

Also removed the duplicate config group

BlueSoapTurtle committed:
Merge branch 'master' into add-order-count-infobox

BlueSoapTurtle commented:
Rebased, ready for review or merge

BlueSoapTurtle committed:
improve 'by station' potion order sorting (#80)

BlueSoapTurtle committed:
Cleanup code

- renamed updateInfoboxes to refreshInfoboxes
- only call refreshInfoboxes when the correct config key is changed
- moved the varp check to the bottom since it's separate from the varbits

BlueSoapTurtle committed:
Merge branch 'master' into add-order-count-infobox

BlueSoapTurtle committed:
Fix some spaces and move out of ui package

BlueSoapTurtle commented:
Had to rebase so made a few changes:
- renamed updateInfoboxes to refreshInfoboxes
- only call refreshInfoboxes when the correct config key is changed
- moved the varp check to the bottom since it's separate from the varbits
- moved OrdersFulfilledInfoBox out of the ui package

Ready again for review or merge
